tfsdk/schema.go

```diff
@@ -208,3 +210,120 @@ func (s Schema) validate(ctx context.Context, req ValidateSchemaRequest, resp *V
 		})
 	}
 }
+
+// modifyAttributePlans runs all AttributePlanModifiers in all schema attributes
+func (s Schema) modifyAttributePlans(ctx context.Context, req ModifySchemaPlanRequest, resp *ModifySchemaPlanResponse) {
+	modifyAttributesPlans(ctx, s.Attributes, tftypes.NewAttributePath(), req, resp)
+}
+
+func modifyAttributesPlans(ctx context.Context, attrs map[string]Attribute, path *tftypes.AttributePath, req ModifySchemaPlanRequest, resp *ModifySchemaPlanResponse) {
+	for name, nestedAttr := range attrs {
+		attrPath := path.WithAttributeName(name)
+		attrPlan, diags := req.Plan.GetAttribute(ctx, attrPath)
+		resp.Diagnostics = append(resp.Diagnostics, diags...)
+		if diagnostics.DiagsHasErrors(diags) {
+			return
+		}
+		nestedAttrReq := ModifyAttributePlanRequest{
+			AttributePath: attrPath,
+			Config:        req.Config,
+			State:         req.State,
+			Plan:          req.Plan,
+			ProviderMeta:  req.ProviderMeta,
+		}
+		nestedAttrResp := &ModifyAttributePlanResponse{
+			AttributePlan: attrPlan,
+			Diagnostics:   resp.Diagnostics,
+		}
+
+		nestedAttr.modifyPlan(ctx, nestedAttrReq, nestedAttrResp)
+		if nestedAttrResp.RequiresReplace {
+			resp.RequiresReplace = append(resp.RequiresReplace, attrPath)
+		}
+
+		setAttrDiags := resp.Plan.SetAttribute(ctx, attrPath, nestedAttrResp.AttributePlan)
+		resp.Diagnostics = append(resp.Diagnostics, setAttrDiags...)
+		if diagnostics.DiagsHasErrors(setAttrDiags) {
+			return
+		}
+		resp.Diagnostics = nestedAttrResp.Diagnostics
```

Comment:
Is this last line expected? I _think_ its just extraneous since they should point to the same address, but it could be  confusing as it looks like it might be removing `SetAttribute` diagnostics.